File: RedSheet.DbEntities/Models/NanoOurObjective.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("NanoOurObjectives", Schema = "dbo")]
	[Serializable]
	public partial class NanoOurObjective  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int NanoOurObjectiveId { get; set; } 
 
	[Required] 
	[MaxLength(150)]
	public string NanoOurObjectiveValue { get; set; } 
 
	[RelationshipTableAttribue("NanoScopeToNegotiateObjectives","dbo")]
	//Relationships
	public int NanoScopeToNegotiateObjectiveId { get; set; } 
 
	[ForeignKey("NanoScopeToNegotiateObjectiveId")]
	public NanoScopeToNegotiateObjective NanoScopeToNegotiateObjective { get; set; } 

	public NanoOurObjective()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/nano-our-objective.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { NanoScopeToNegotiateObjective,  vNanoOurObjectiveRecord  } from './'
export class NanoOurObjective {
    constructor(nanoOurObjective?: NanoOurObjective  | vNanoOurObjectiveRecord )  {
        let properties = [ "nanoOurObjectiveId", "nanoOurObjectiveValue", "nanoScopeToNegotiateObjectiveId",];
        for (let property of properties)
            if (nanoOurObjective && nanoOurObjective[property])
                this[property] = nanoOurObjective[property];
    }
 
	nanoOurObjectiveId : number =   0 ;
 
    @required()
    @maxLength(150)
	nanoOurObjectiveValue : string =   undefined;
 
    @range(0,2147483647)
	nanoScopeToNegotiateObjectiveId : number =   undefined;
	nanoScopeToNegotiateObjective : NanoScopeToNegotiateObjective  ;


}
